lib/src/yaml_node.dart

```diff
@@ -160,7 +162,7 @@ class YamlScalar extends YamlNode {
   /// [sourceUrl] is passed, it's used as the [SourceSpan.sourceUrl].
   ///
   /// [sourceUrl] may be either a [String], a [Uri], or `null`.
-  YamlScalar.wrap(this.value, {sourceUrl}) : style = ScalarStyle.ANY {
+  YamlScalar.wrap(this.value, {sourceUrl, this.style = ScalarStyle.ANY}) {
     _span = NullSpan(sourceUrl);
```

Comment:
Given that `style` has a default value, is there any reason to allow passing in `null` as `style`?
```suggestion
    ArgumentError.checkNotNull(style, 'style');
    _span = NullSpan(sourceUrl);
```